kobo/apps/superuser_stats/views.py

```diff
@@ -22,40 +31,227 @@ def _base_filename_to_full_filename(base_filename, username):
+        f'<b>Example:</b><br>'
+        f'<a href="{url}?start_date=2020-01-31&end_date=2021-02-28">'
+        f'  {url}?start_date=2020-01-31&end_date=2021-02-28'
+        f'</a>'
+        f'</body></html>'
+    )
+
+    return HttpResponse(template_ish)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def continued_usage_report(request):
+    """
+    Tracks users usage over a given year with the last day being set by
+    'end_date'
+    """
+    today = timezone.now().date()
+    base_filename = 'continued-usage-report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        today,
+        timezone.now().microsecond
+    )
+
+    # Get the date filters from the query and set defaults
+    tomorrow = today + timedelta(days=1)
+    end_date = request.GET.get('end_date', tomorrow)
+
+    # Generate the CSV file
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username)
+    generate_continued_usage_report.delay(filename, end_date)
+
+    url = f"{KOBOFORM_URL}{reverse('superuser_stats:continued_usage_report')}"
+    template_ish = (
+        f'<html><head><title>Continued usage report</title></head>'
+        f'<body>Your report is being generated. Once finished, it will be '
+        f'available at <a href="{base_filename}">{base_filename}</a>.<br>'
+        f'If you receive a 404, please refresh your browser periodically until '
+        f'your request succeeds.<br><br>'
+        f'To select a date range, add a <code style="background: lightgray">?</code> at the end of the URL and set the '
+        f'<code style="background: lightgray">end_date</code> parameter to <code style="background: lightgray">YYYY-MM-DD</code>.<br><br>'
+        f'<b>Example:</b><br>'
+        f'<a href="{url}?end_date=2021-02-28">'
+        f'  {url}?end_date=2021-02-28'
+        f'</a>'
+        f'</body></html>'
+    )
+
+    return HttpResponse(template_ish)
+
+
+@user_passes_test(lambda u: u.is_superuser)
+def domain_report(request):
+    """
+    Generates a report which counts the amount of users
+    with the same email address domain
+    """
+    # Generate the file basename
+    today = timezone.now().date()
+    base_filename = 'domain-report_{}_{}_{}.csv'.format(
+        re.sub('[^a-zA-Z0-9]', '-', request.META['HTTP_HOST']),
+        today,
+        timezone.now().microsecond
+    )
+
+    # Get the date filters from the query and set defaults
+    start_date = request.GET.get(
+        'start_date',
+        f'{today.year}-{today.month}-{today.day}'
+    )
+    tomorrow = timezone.now() + timedelta(days=1)
+    end_date = request.GET.get(
+        'end_date',
+        f'{tomorrow.year}-{tomorrow.month}-{tomorrow.day}'
+    )
+
+    # Generate the CSV file
+    filename = _base_filename_to_full_filename(
+        base_filename, request.user.username)
+    generate_domain_report.delay(filename, start_date, end_date)
```

Comment:
This also doesn't seem to be working as the export only included the headers without any user data.

Reply:
Once again the date range. 

> The default date range is for today, but submissions count will not be 0 unless it includes the range includes first of the month.

I think it's misleading. As @joshuaberetta said, if there is no data for the data range, it returns nothing, not `0`. We should change the explanation in the HTML template.

Reply:
If I try getting these stats in the shell there is data there:
```python
>>> pprint(Counter([e['email'].split('@')[-1] for e in User.objects.all().values('email') if e['email']]))
Counter({'kobotoolbox.org': 5,
         'gmail.com': 2,
         'kobotoolbox.com': 1,
         'j.com': 1,
         'yahoo.com': 1,
         's.com': 1,
         'example.com': 1})
```

Reply:
Ah nice 👍 
There's a strange line included in the export:


Email Domain | Users | Projects | Submissions
-- | -- | -- | --
!!invalid: | 4 | 0 | 0

Reply:
It means, email addresses do not match `xxx@domain.ltd`

Reply:
Should we change it to something else? That looks more like an error occurred

Reply:
Any suggestion for what to use instead?

Reply:
Maybe just "invalid domain"? Or "blank" since that's what seems the be the reason:
```python
>>> pprint(Counter([e['email'].split('@')[-1] for e in User.objects.all().values('email')]))
Counter({'kobotoolbox.org': 5,
         '': 4,
         'gmail.com': 2,
         'kobotoolbox.com': 1,
         'j.com': 1,
         'yahoo.com': 1,
         'example.com': 1,
         's.com': 1})
```